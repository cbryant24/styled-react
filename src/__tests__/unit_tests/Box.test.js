import React from 'react'
import { shallow } from 'enzyme';
import {
  Div
} from '@cbryant24/styled-react';
import 'jest-styled-components';
import { act } from 'react-dom/test-utils';
import wait from 'waait';
// import { updateComponent } from '../utils';

let component;

async function updateComponent(component, amount = 0) {
	await act(async () => {
		await wait(amount);
		component.update();
	});
}

describe('styled-box', () => {
  beforeEach(async () => {
    component = shallow(
      <Div width="75%"></Div>
    )

    await updateComponent(component);
  });
  it('has a div with margin 50%', () => {
    expect(component).toMatchSnapshot()
    expect(component).toHaveStyleRule('width', '75%')
  })
});