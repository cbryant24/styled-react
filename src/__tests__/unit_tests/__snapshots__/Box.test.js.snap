// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled-box has a div with margin 50% 1`] = `
.c0 {
  box-sizing: border-box;
  min-width: 0;
  width: 75%;
}

<Component
  className="c0"
  isA="div"
  width="75%"
/>
`;
